replication/binlogsyncer.go

```diff
@@ -509,7 +515,14 @@ func (b *BinlogSyncer) writeBinlogDumpCommand(p mysql.Position) error {
 	binary.LittleEndian.PutUint32(data[pos:], p.Pos)
 	pos += 4
 
-	binary.LittleEndian.PutUint16(data[pos:], b.cfg.DumpCommandFlag)
+	dumpCommandFlag := b.cfg.DumpCommandFlag
+	if b.cfg.MariaDBDynamicLogPos && b.cfg.Flavor == mysql.MariaDBFlavor {
+		// Add BINLOG_SEND_ANNOTATE_ROWS_EVENT flag when MariaDBDynamicLogPos is enabled.
+		// This ensures the server sends ANNOTATE_ROWS_EVENT events which are needed
+		// for correct LogPos calculation in MariaDB 11.4+, where some events have LogPos=0.
```

Comment:
Do you have some manual test result to make sure only this flag is enough? Please add some explanation in PR description.

Reply:
Don't worry. I think because this fix indeed fixed your problem, we can merge it and leave more enhancement to future PR.

And because it's possible (not likely, though) we will change to another implementation and break properties from this PR, can you add some unit tests to prevent it breaking unintentionally?